noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr

```diff
@@ -51,24 +53,14 @@ contract Crowdfunding {
     }
     // docs:end:init
 
-    // docs:start:deadline
-    // docs:start:deadline-header
-    #[aztec(public)]
-    #[aztec(internal)]
-    #[aztec(view)]
-    fn _check_deadline() {
-        // docs:end:deadline-header
-        let deadline = storage.deadline.read();
-        assert(context.timestamp() < deadline, "Deadline has passed");
-    }
-    // docs:end:deadline
-
     // docs:start:donate
-    // docs:start:call-check-deadline
     #[aztec(private)]
     fn donate(amount: u64) {
-        // 1) Check that the deadline has not passed
-        Crowdfunding::at(context.this_address())._check_deadline().enqueue_view(&mut context);
+        // 1) Check that the deadline has not passed --> we do that via the router contract to conceal which contract
+        // is performing the check.
+        // docs:start:call-check-deadline
+        let deadline = storage.deadline.read_private();
+        Router::at(ROUTER_ADDRESS).check_timestamp(deadline, Comparator.GT).call(&mut context);
```

Comment:
I had issue remembering on what side is the passed in argument in the comparison. I renamed that arg as `lhs` and put it on the left so that it's more easily remembered.

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/8384